docs/proposals/service-discovery/README.md

```diff
@@ -0,0 +1,171 @@
+- Assuming the pod is exported as the service `foo`
+- Another pod that wishes to access it on the same cluster will simply call `foo` and Kubernetes will bind to the correct one
+- If that pod is scheduled to another cluster, the original service discovery will fail as there's no service by the name `foo`
+- To find the original pod, the other pod is required to know it is in another cluster and use `derived-foo` to work properly
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.
+-->
+
+### Goals
+
+- Remove the "derived-" prefix from the service
+- User-friendly and native service discovery
+
+### Non-Goals
+
+- Multi cluster connectivity
+
+## Proposal
+
+Following are flows to support the service import proposal:
+
+1. `Deployment` and `Service` are created on cluster member1 and the `Service` imported to cluster member2 using `ServiceImport` (described below as [user story 1](#story-1))
+2. `Deployment` and `Service` are created on cluster member1 and both propagated to cluster member2. `Service` from cluster member1 is imported to cluster member2 using `ServiceImport` (described below as [user story 2](#story-2))
+
+The proposal for this flow is what can be referred to as local-and-remote service discovery. The options could be:
+
+1. **Local** only - In case there's a local service by the name `foo` Karmada never attempts to import the remote service and doesn't create an `EndPointSlice`
+2. **Local** and **Remote** - Users accessing the `foo` service will reach either member1 or member2
+3. **Remote** only - in case there's a local service by the name `foo` Karmada will remove the local `EndPointSlice` and will create an `EndPointSlice` pointing to the other cluster (e.g. instead of resolving member2 cluster is will reach member1)
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a Kubernetes cluster member,
+I want to access a service from another cluster member,
+So that I can communicate with the service using its original name.
+
+**Background**: The Service named `foo` is created on cluster member1 and imported to cluster member2 using `ServiceImport`.
+
+**Scenario**:
+
+1. Given that the `Service` named `foo` exists on cluster member1
+2. And the `ServiceImport` resource is created on cluster member2, specifying the import of `foo`
+3. When I try to access the service inside member2
+4. Then I can access the service using the name `foo.myspace.svc.cluster.local`
+5. And the import process does not prefix the service name with `derived-`
+6. And the `EndPointSlice` associated with the imported service is prefixed with `import-`
```

Comment:
I think these two points are related to specific implementation and can be explained in the details of the implementation.